vmdb/app/models/zone.rb

```diff
@@ -130,7 +130,11 @@ def host_ids
 
   def hosts
     MiqPreloader.preload(self, :ext_management_systems => :hosts)
-    self.ext_management_systems.collect(&:hosts).flatten
+    ext_management_systems.collect(&:hosts).flatten
```

Comment:
All the `.collect(&:key).flatten` calls can be changed to use flat_map: `ext_management_systems.flat_map(&:hosts)`.